File: 2364-count-number-of-bad-pairs/2364-count-number-of-bad-pairs.py
class Solution:
    def countBadPairs(self, nums: List[int]) -> int:
        mp=dict()
        n=len(nums)
        count=n*(n-1)//2
        for i in range(0,n):
            if nums[i]-i in mp:
                count-=mp[nums[i]-i]
            if nums[i]-i not in mp:
                mp[nums[i]-i]=0
            mp[nums[i]-i]+=1    
        return count    
        

File: 2364-count-number-of-bad-pairs/2364-count-number-of-bad-pairs.cpp
class Solution {
public:
    long long countBadPairs(vector<int>& nums) {
        unordered_map<long long ,long long>mp;
        int n=nums.size();
        long long count=1ll*n*(n-1)/2;
        for(int i=0;i<n;i++)
        {
            if(mp.find(nums[i]-i)!=mp.end())
            {
                count-=1ll*mp[nums[i]-i];
            }
            mp[nums[i]-i]++;
        }
        return count;
    }
};